beps/0004-scaffolder-task-idempotency/README.md

```diff
@@ -224,6 +222,41 @@ It's going look like:
 }
 ```
 
+or a failed attempt as:
+
+```json
+{
+  "repo.creation": {
+    "status": "failed",
+    "reason": "Namespace is not valid"
+  }
+}
+```
+
+DatabaseTaskStore will provide two extra methods `saveTaskState` and `getTaskState`. The type of state in API will be
+represented as `JsonObject`.
+
+Task state will be stored in the extra column `state` in the table `tasks` with the next structure:
+
+```json
+{
+  "state": {
```

Comment:
I think that we still want the checkpoints key in this that stores all the checkpoints and their status.


`
```suggestion
  "state": {
    "checkpoints": {
     ...
```